comp/core/secrets/secretsimpl/secrets.go

```diff
@@ -401,12 +415,24 @@ func (r *secretResolver) Refresh() error {
 		secretResponse, err = r.fetchSecret(newHandles)
 	}
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	// when Refreshing secrets, only update what the allowlist allows
-	r.processSecretResponse(secretResponse, true)
-	return nil
+	// when Refreshing secrets, only update what the allowlist allows by passing `true`
+	refreshResult := r.processSecretResponse(secretResponse, true)
+
+	// render a report
+	t := template.New("secret_refresh")
+	t, err = t.Parse(secretRefreshTmpl)
+	if err != nil {
+		return "", err
+	}
+	b := new(strings.Builder)
+	err = t.Execute(b, refreshResult)
+	if err != nil {
+		return "", err
+	}
+	return b.String(), nil
```

Comment:
I don't know if the backend for refresh secrets should also handle the rendering of the template.
If, in the future, we want to allow customers to get a JSON output, we would have to pass many args from the CLI to the backend.

What do you think of moving the rendering side of things to the CLI?

Reply:
I'm torn because there's already rendering logic here. Generally I prefer to keep UI in the frontend, but until there's a stronger reason to separate the rendering code into two different places, I think it's fine staying here for now.